src/main/java/it/rate/webapp/controllers/api/ImageRestController.java

```diff
@@ -25,4 +27,30 @@ public ResponseEntity<?> getImage(@PathVariable String id) {
       return ResponseEntity.internalServerError().build();
     }
   }
+
+  @PostMapping("/new-interest-image")
+  @PreAuthorize("@permissionService.canCreateInterest()")
+  public ResponseEntity<?> uploadNewInterestImage(@RequestParam("picture") MultipartFile file) {
+
+    try {
+      return ResponseEntity.ok()
+          .body(new ImageUploadResponseDTO(googleImageService.saveImage(file)));
+    } catch (IOException e) {
+      return ResponseEntity.internalServerError().build();
+    }
+  }
+
+  @PutMapping("/interests/{interestId}/edit")
+  @PreAuthorize("@permissionService.manageCommunity(#interestId)")
+  public ResponseEntity<?> changeInterestImage(
+      @RequestParam("picture") MultipartFile file, @PathVariable Long interestId) {
+
+    try {
+      return ResponseEntity.ok()
+          .body(
+              new ImageUploadResponseDTO(googleImageService.changeInterestImage(interestId, file)));

```

Comment:
It's more future proof if getting interest by id is moved to this method where it is safe due to preauthorization and obtained internet is then fed to changeInterestImage method.

Reply:
Getting interest in controller done.